xontrib/jedi.xsh

```diff
@@ -1,51 +1,138 @@
-"""Jedi-based completer for Python-mode."""
-import builtins
-import importlib
+"""Use Jedi as xonsh's python completer."""
+import itertools
 
+import xonsh
 from xonsh.lazyasd import lazyobject, lazybool
+from xonsh.completers.tools import (
+    get_filter_function,
+    get_ptk_completer,
+    RichCompletion,
+)
 
 
 __all__ = ()
 
 
-@lazybool
-def HAS_JEDI():
-    """``True`` if `jedi` is available, else ``False``."""
-    spec = importlib.util.find_spec('jedi')
-    return (spec is not None)
+try:
+    import jedi
+except ImportError:
+    raise SystemExit
```

Comment:
I don't think xontribs should kill xonsh...

Reply:
At least not on import :wink:

Reply:
It won't, `SystemExit` only exits the script (from #3241)
Is this bad practice?

Reply:
Yeah, but xontribs are not scripts, exactly.  They can be loaded on REPL startup as well.  This is not what we want, I don't think.  I think there should be a different escape hatch, like a function which returns None on load failure, for example

Reply:
OK so apparently `SystemExit` really exits the xontribs process (or the xonsh process when the xontribs command is in xonshrc).

But if some other exception is raised (say `ModuleNotFoundError`) it's printed to the screen, xontribs catches it and recognizes the xontrib failed to load.
So I left it as simply importing jedi in `jedi.xsh`, and I think the result is acceptable:

![image](https://user-images.githubusercontent.com/18242949/88599091-981bda00-d073-11ea-952b-56ebabee7928.png)

Reply:
Yep! I think that is good enough!